src/Donations/Micros/OnBehalf.tsx

```diff
@@ -0,0 +1,141 @@
+        />
+      </div>
+      {onBehalf && onBehalfDonor.firstName === "" ? (
+        <div className="on-behalf-container-form-container mt-10">
+          <Grid container spacing={3}>
+            <Grid item xs={6}>
+              <MaterialTextField
+                name="firstName"
+                inputRef={register({ required: true })}
+                label={t("firstName")}
+                variant="outlined"
+              />
+              {errors.firstName && (
+                <span className={"form-errors"}>{t("firstNameRequired")}</span>
+              )}
+            </Grid>
+            <Grid item xs={6}>
+              <MaterialTextField
+                name="lastName"
+                inputRef={register({ required: true })}
+                label={t("lastName")}
+                variant="outlined"
+              />
+              {errors.lastName && (
+                <span className={"form-errors"}>{t("lastNameRequired")}</span>
+              )}
+            </Grid>
+            {shouldNotify ? (
+              <Grid item xs={12}>
+                <div className="d-flex row justify-content-between mb-10">
+                  <p>{t("onBehalfNotification")}</p>
+                  <button
+                    onClick={() => setShouldNotify(false)}
+                    className={"singleGiftRemove"}
+                  >
+                    {t("removeRecipient")}
+                  </button>
+                </div>
+                <MaterialTextField
+                  name="email"
+                  inputRef={register({
+                    pattern:
+                      /^([a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?)$/i,
+                  })}
+                  label={t("email")}
+                  variant="outlined"
+                />
+                {errors.email && (
+                  <span className={"form-errors"}>
+                    {t("inValidField", {
+                      fieldName: "Email",
+                    })}
+                  </span>
+                )}
+              </Grid>
+            ) : (
+              <Grid item xs={12}>
+                <div className={"form-field"}>
+                  <button
+                    onClick={() => setShouldNotify(true)}
+                    className={"addEmailButton"}
+                  >
+                    {t("addEmail")}
+                  </button>
+                </div>
+              </Grid>
+            )}
+          </Grid>
+          <button
+            onClick={handleSubmit(onSubmit)}
+            className="primary-button w-100 mt-30"
+          >
+            {t("continue")}
+          </button>
+        </div>
+      ) : (
+        <></>
+      )}
+    </div>
+  ) : (
+    <div className="on-behalf-donor-info mt-10">
+      <p>
+        {t("donationOnBehalfOf")}&nbsp;
+        {onBehalfDonor.firstName + " " + onBehalfDonor.lastName}
```

Comment:
It would be nicer for translators if the name could be a parameter for the translation resource, so the translation can decide about where to place the name within the resource.

Reply:
Done